### PR TITLE
fix(norm): widen address arithmetic to int64 for large contiguous inputs > 2**31 elements

### PR DESCRIPTION
**Pull request by bkryu**
## 📌 Description

Fix cudaErrorIllegalAddress in the cute-DSL norm kernels (`rmsnorm`, `gemma_rmsnorm`, `rmsnorm_quant`, `fused_add_rmsnorm`, `gemma_fused_add_rmsnorm`, `fused_add_rmsnorm_quant`, `layernorm`, and `qk_rmsnorm`) when the input tensor's flat element count exceeds `2**31`. 

**Root Cause**
The compact (contiguous) compile path bakes the row stride in as a Python constexpr integer, and M / B / N were declared as `Int32` everywhere. The row coord materialized into the offset comes from `cute.arch.block_idx()`, which is intrinsically `int32`, so the compiler emits `row * H` in `int32` and overflows whenever `M * H > INT32_MAX`. For shape (175000, 12288) fp16 (174999 * 12288 ≈ 2.15e9) this manifests as `cudaErrorIllegalAddress` on the first `cudaStreamSynchronize`. The strided compile path was already safe because its row stride was already a dynamic `sym_int64`, but the contiguous path was not.

A secondary bug existed in `flashinfer/norm/utils.py::get_ptr_as_int64`, which truncated its offset parameter to Int32 before adding it to the base pointer — so even a correctly-computed `int64` element offset was being downcast in the rmsnorm-quant kernels' manual FP8 store path.

## 🔍 Related Issues

#3391

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [x] I have installed the hooks with `pre-commit install`.
- [x] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [ ] Tests have been added or updated as needed.
- [ ] All tests are passing (`unittest`, etc.).

## Summary by CodeRabbit

* **Bug Fixes**
  * Resolved integer-overflow issues in normalization (RMSNorm, LayerNorm, fused Add+RMSNorm and FP8-quantized variants) for very large tensors, preventing illegal memory accesses and ensuring correct addressing for huge row counts.
  * Forced a safe compilation/execution path for contiguous inputs when dimensions exceed 32-bit limits.

* **Tests**
  * Added regression tests that detect contiguous overflow scenarios across normalization variants, with VRAM-aware gating and spot-check validation.

[Review Change Stack](https://app.coderabbit.ai/change-stack/flashinfer-ai/flashinfer/pull/3392?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: flashinfer/norm/kernels/fused_add_rmsnorm.py [hunk 1]
```diff
@@ -753,7 +753,7 @@ def kernel(
                         get_ptr_as_int64(
                             mY,
                             cute.crd2idx(
-                                (Int32(actual_row), Int32(abs_col)), mY.layout
+                                (Int64(actual_row), Int32(abs_col)), mY.layout
```
[the remark this answers is not in the document]
Comment (bkryu) [reply flashinfer/norm/kernels/fused_add_rmsnorm.py:759]: Good catch — agreed. With M widened to Int64 it's now legal for a caller to pass M > INT32_MAX whenever rows_per_block > 1 (e.g. rows_per_block = 2 with a near-max grid_x), and in that regime bidx * rows_per_block overflows int32 before we ever get to the crd2idx cast. The actual_row < M bound check on the next line is also wrong in that case — comparing an int32-overflowed value against an int64 M.
  
Updated both RMSNormQuantKernel and FusedAddRMSNormQuantKernel to compute actual_row in int64 by casting bidx up

File: flashinfer/norm/kernels/rmsnorm.py [hunk 2]
```diff
@@ -978,7 +978,7 @@ def kernel(
                         get_ptr_as_int64(
                             mY,
                             cute.crd2idx(
-                                (Int32(actual_row), Int32(abs_col)), mY.layout
+                                (Int64(actual_row), Int32(abs_col)), mY.layout
```
[the remark this answers is not in the document]
Comment (bkryu) [reply flashinfer/norm/kernels/rmsnorm.py:984]: Ditto as above.

File: tests/utils/test_norm.py [hunk 3]
```diff
@@ -471,6 +471,244 @@ def test_fused_add_rmsnorm_quant_int64_stride():
...
+    # input fp16 (2 B) + fp8 output (1 B).
+    _skip_if_low_vram_for_overflow(extra_bytes_per_elem=3.0)
+    quant_scale = 1.0
+    x = torch.randn(_OVERFLOW_M, _OVERFLOW_H, dtype=_OVERFLOW_DTYPE, device="cuda")
+    w = torch.randn(_OVERFLOW_H, dtype=_OVERFLOW_DTYPE, device="cuda")
+    assert x.is_contiguous()
+
+    y = torch.empty(_OVERFLOW_M, _OVERFLOW_H, dtype=torch.float8_e4m3fn, device="cuda")
+    flashinfer.norm.rmsnorm_quant(y, x, w, torch.tensor(quant_scale, device="cuda"))
+    torch.cuda.synchronize()
```
[the remark this answers is not in the document]
Comment (bkryu) [reply tests/utils/test_norm.py:560]: None of the existing FP8-output norm tests in tests/utils/test_norm.py apply an architecture skip. A simple fp8 conversion should be fine on all architectures.

**Review by kahyunnam — APPROVED** [review]
LGTM
